buildSrc/src/main/kotlin/pklJavaLibrary.gradle.kts

```diff
@@ -80,9 +101,62 @@ val workAroundKotlinGradlePluginBug by
     }
   }
 
+val truffleJavacArgs =
+  listOf(
+    // TODO: determine correct limits for Truffle specializations
+    // (see https://graalvm.slack.com/archives/CNQSB2DHD/p1712380902746829)
+    "-Atruffle.dsl.SuppressWarnings=truffle-limit"
+  )
+
+val jpmsJavacArgs =
+  listOf(
+    // Required for use of `sun.misc.Unsafe` in Truffle.
+    "--add-modules=jdk.unsupported"
+  )
+
+val javacArgsProvider = CommandLineArgumentProvider { jpmsJavacArgs }
+
 tasks.compileJava {
+  javaCompiler = info.javaCompiler
   dependsOn(workAroundKotlinGradlePluginBug)
-  // TODO: determine correct limits for Truffle specializations
-  // (see https://graalvm.slack.com/archives/CNQSB2DHD/p1712380902746829)
-  options.compilerArgs.add("-Atruffle.dsl.SuppressWarnings=truffle-limit")
+  options.compilerArgs.addAll(truffleJavacArgs)
 }
+
+tasks.withType<JavaCompile>().configureEach {
+  val jvmTarget = JavaVersion.toVersion(info.jvmTarget)
+  javaCompiler = info.javaCompiler
+  sourceCompatibility = jvmTarget.majorVersion
+  targetCompatibility = jvmTarget.majorVersion
+
+  options.compilerArgumentProviders.add(javacArgsProvider)
+}
+
+tasks.withType<JavaExec>().configureEach { jvmArgs(jpmsJavacArgs) }
+
+fun Test.configureJdkTestTask(launcher: Provider<JavaLauncher>) {
+  useJUnitPlatform()
+  minHeapSize = "128M"
+  maxHeapSize = "512M"
+  forkEvery = 200
+  maxParallelForks = 4
```

Comment:
these seem to speed up tests for me, but we should see how it performs in CI, and I can easily drop these